File: CP/LeetCode/CPP/1.cpp
// 1. Two Sum

#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        vector<int> result;
        int size = (int)nums.size();
        for (int i = 0; i < size; i++)
        {
            int temp = target - nums[i];
            for (int j = i + 1; j < size; j++)
            {
                if (nums[j] == temp)
                {
                    result.push_back(i);
                    result.push_back(j);
                    return result;
                }
            }
        }
        return result;
    }
};

int main()
{
    Solution sol;
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;
    vector<int> result = sol.twoSum(nums, target);
    int size = (int)result.size();
    for (int i = 0; i < size; i++)
    {
        cout << result[i] << " ";
    }
}

File: CP/LeetCode/Java/1.java
// 1. Two Sum

import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(final int[] nums, final int target) {
        int[] result = new int[2];
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(target - nums[i])) {
                result[0] = map.get(target - nums[i]);
                result[1] = i;
                return result;
            }
            map.put(nums[i], i);
        }
        return result;
    }

    public static void main(final String[] args) {
        final Solution solution = new Solution();
        final int[] nums = { 2, 7, 11, 15 };
        final int target = 9;
        final int[] result = solution.twoSum(nums, target);
        System.out.println(result[0] + " " + result[1]);
    }
}
